bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/views/DiagnosenDisplay.java

```diff
@@ -330,6 +336,43 @@ public String getText(Object element) {
 				return StringTool.leer;
 			}
 		});
+		ViewerComparator codeComparator = new ViewerComparator() {
+			@Override
+			public int compare(Viewer viewer, Object e1, Object e2) {
+				if (e1 instanceof IDiagnosisReference && e2 instanceof IDiagnosisReference) {
+					IDiagnosis b1 = (IDiagnosis) e1;
+					IDiagnosis b2 = (IDiagnosis) e2;
+					return b1.getCode().compareTo(b2.getCode());
+				}
+				return 0;
+			}
+		};
+		col.getColumn().addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (viewer.getComparator() == codeComparator) {
+					isCodeSortedAscending = !isCodeSortedAscending;
+					viewer.setComparator(new ViewerComparator() {
+						private boolean isCodeSortedAscending;
+
+						@Override
+						public int compare(Viewer viewer, Object e1, Object e2) {
+							if (e1 instanceof IDiagnosisReference && e2 instanceof IDiagnosisReference) {
+								IDiagnosis b1 = (IDiagnosis) e1;
+								IDiagnosis b2 = (IDiagnosis) e2;
+								int result = b1.getCode().compareTo(b2.getCode());
+								return isCodeSortedAscending ? result : -result;
+							}
+							return 0;
+						}
+					});
+				} else {
+					viewer.setComparator(codeComparator);
+					isCodeSortedAscending = true; // Zurücksetzen auf aufsteigende Sortierung
+				}
+				viewer.refresh();
+			}
+		});
```

Comment:
Hier wird immer noch beim Ändern der Richtung, das ViewerComparator Objekt neu erzeugt und aus getauscht.

Bitte schau folgenden Beispiel Code an. Dabei wird die Reihenfolge als Data Eigenschaft des Viewer hinterlegt, und der ViewerComparator liest das aus und ändert entsprechend die Reihenfolge.
<pre>
		ViewerComparator codeComparator = new ViewerComparator() {
			@Override
			public int compare(Viewer viewer, Object e1, Object e2) {
				if (e1 instanceof IDiagnosisReference && e2 instanceof IDiagnosisReference) {
					IDiagnosis b1 = (IDiagnosis) e1;
					IDiagnosis b2 = (IDiagnosis) e2;
					int result = b1.getCode().compareTo(b2.getCode());
					return getSortedAscending(viewer) ? result : -result;
				}
				return 0;
			}

			private boolean getSortedAscending(Viewer viewer) {
				if (viewer.getData("codeSortAscending") != null) {
					return (Boolean) viewer.getData("codeSortAscending");
				}
				return true;
			}
		};

		col.getColumn().addSelectionListener(new SelectionAdapter() {
			@Override
			public void widgetSelected(SelectionEvent e) {
				if (viewer.getComparator() != null) {
					if (viewer.getData("codeSortAscending") != null) {
						viewer.setData("codeSortAscending", !(Boolean) viewer.getData("codeSortAscending"));
					} else {
						viewer.setData("codeSortAscending", Boolean.FALSE);
					}
					viewer.refresh();
				}
			}
		});
</pre>
Ich habe den Code nicht fertig getestet, aber in der Art sollte das klappen. Bei den anderen Splaten muss dann nur entsprechend der Key ("codeSortAscending") unter dem die Reihenfolge abgelegt ist ausgetauscht werden.